clang-tools-extra/clangd/tool/ClangdMain.cpp

```diff
@@ -694,13 +698,20 @@ class FlagsConfigProvider : public config::Provider {
       BGPolicy = Config::BackgroundPolicy::Skip;
     }
 
+    if (PlaceholderOption == Config::ArgumentListsOption::Delimiters) {
+      ArgumentLists = PlaceholderOption;
+    }
+
     Frag = [=](const config::Params &, Config &C) {
       if (CDBSearch)
         C.CompileFlags.CDBSearch = *CDBSearch;
       if (IndexSpec)
         C.Index.External = *IndexSpec;
       if (BGPolicy)
         C.Index.Background = *BGPolicy;
+      if (ArgumentLists && C.Completion.ArgumentLists ==
```

Comment:
For reasons mentioned above, no need for the second condition here (at this point, the config provider for the config file has not **yet** assigned anything to `C`; when it does run later, it will overwrite the field if needed)